src/image/hfe.c

```diff
@@ -375,11 +378,66 @@ static bool_t hfe_write_track(struct image *im)
             + (im->cur_track & 1) * 256
             + batch_off - im->hfe.write_batch.off
             + (off & 255);
-        for (i = 0; i < nr; i++)
+
+        i = 0;
+
+        if (im->hfe.fresh_seek && is_v3 && (off & 255) >= 1) {
```

Comment:
What if an opcode straddles the 256-byte boundary? You can't just not handle that particular case, right?

Reply:
Correct. But that's only if the write starts near a 256-byte boundary and the opcode spans the block. Once the write starts it handles the boundaries fine.

Thinking about this again, it'd better to shift the start of the write by 1-2 bytes so it starts no earlier than the 3rd byte of the block, and then do this processing.

Reply:
Yes this was my thought, jogging the write start is a better idea.